daml/Marketplace/Custody.daml

```diff
@@ -121,21 +121,21 @@ template Service
         with
           creditAccountRequestCid : ContractId CreditAccountRequest
         do
-          CreditAccountRequest{..} <- fetchAndArchive creditAccountRequestCid
+          CreditAccountRequest{accountId; asset} <- fetchAndArchive creditAccountRequestCid
           exerciseByKey @AssetSettlementRule accountId AssetSettlement_Credit with ctrl = provider; ..
 
       nonconsuming DebitAccount : Asset
         with
           debitAccountRequestCid : ContractId DebitAccountRequest
         do
-          DebitAccountRequest{..} <- fetchAndArchive debitAccountRequestCid
+          DebitAccountRequest{accountId; depositCid} <- fetchAndArchive debitAccountRequestCid
           exerciseByKey @AssetSettlementRule accountId AssetSettlement_Debit with ..
```

Comment:
Right, so this is a good example of what I meant before, as well as the next change on line 138 ... here we can change this to:

```
data DebitAccountRequest { accountId: Id, request : AssetSettlement_debit }

...

      nonconsuming DebitAccount : Asset
        with
          debitAccountRequestCid : ContractId DebitAccountRequest
        do
          DebitAccountRequest{accountId, request} <- fetchAndArchive debitAccountRequestCid
          exerciseByKey @AssetSettlementRule accountId request
```
or even

I don't want to dwell on this - you're spending a lot of time doing this refactoring, but I just wanted to make clear what I was suggesting.

Reply:
I've made this change. Looking at other services, it's a bit tricker to know if it makes sense to use it as most choices have more logic than this simple proxy example.